AppController/djinn.rb

```diff
@@ -4695,9 +4713,12 @@ def setup_appengine_application(app, state)
 
       if @app_info_map[app]['nginx'].nil?
         @app_info_map[app]['nginx'] = nginx_app_port
+      end
+      if @app_info_map[app]['nginx_https'].nil?
+        @app_info_map[app]['nginx_https'] = nginx_app_https
```

Comment:
It looks like the port found by `find_lowest_free_port` is ignored when there's a value in the `@app_info_map`. Can't the port be taken before the lock is established?